homeassistant/components/growatt_server/config_flow.py

```diff
@@ -0,0 +1,53 @@
+"""Config flow for growatt server integration."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_USERNAME, CONF_PASSWORD
+from .const import DOMAIN, CONF_PLANT_ID, DEFAULT_NAME
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class GrowattServerConfigFlow(config_entries.ConfigFlow):
+    """Config flow class."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await self.async_step_user(user_input)
+
+    async def _show_form(self, errors=None):
+        """Show the form to the user."""
+        data_schema = vol.Schema(
+            {
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                vol.Optional(CONF_PLANT_ID): str,
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="init", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        if not user_input:
+            return await self._show_form()
+
+        import growattServer
+
+        api = growattServer.GrowattApi()
+        login_response = api.login(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+
+        if not login_response["success"] and login_response["errCode"] == "102":
+            return await self._show_form({"base": "auth_error"})
+        user_id = login_response["userId"]
+
+        if CONF_PLANT_ID not in user_input:
+            plant_info = api.plant_list(user_id)
```

Comment:
Instead of asking the user for a plant ID in the first step, why not add a second step that lists all the plants and allow the user to pick from the dropdown ? You can make the dropdown have readable names using the following format for your data_schema:

```python
data_schema=vol.Schema({
    "plant_id": vol.In({
        12: "Plant A",
        34: "Plant B",
    })
})
```

Reply:
Thank you, that sound like a good idea and i've now implemented it in a way where it only shows the step if you have multiple "plants" on your account.